acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+          in json format, in comparison to no space exists.
+       2. Total number of space should be 1.
+       3. The name of new added space should be the same as specified.
+       4. Total number of subnet CIDR from this new added space should be 1.
+       5. The CIDR data from new added space should be the same as specified.
+    """
+    space_name, subnet_cidr = add_space_with_existing_subnet(
+        client, space_name='testspace')
+    # New added space should exist at this point,
+    # set merge_stderr to False to eliminate noise in output.
+    raw_output = client.get_juju_output(
+        'spaces', '--format', 'json', include_e=False, merge_stderr=False)
+    try:
+        space_output = json.loads(raw_output)
+    except ValueError as e:
+        log.error('New added space cannot be found.')
+        raise e
+    # These 4 validations are in sequence.
+    if len(space_output['spaces'].keys()) != 1:
+        raise JujuAssertionError(
+            'Incorrect number of space(s). Found: {}; Expected: 1.'.format(
+            str(len(space_output['spaces'].keys()))))
+    elif space_output['spaces'].keys()[0] != space_name:
+        raise JujuAssertionError(
+            'Incorrect space name. Found: {}; Expected: {}.'.format(
+            space_output['spaces'].keys()[0], space_name))
+    elif len(space_output['spaces'][space_name].keys()) != 1:
+        raise JujuAssertionError(
+            'Incorrect number of CIDR(s). Found: {}; Expected: 1.'.format(
+            str(len(space_output['spaces'][space_name].keys()))))
+    elif space_output['spaces'][space_name].keys()[0] != subnet_cidr:
+        raise JujuAssertionError(
+            'Incorrect CIDR name. Found: {}; Expected: {}.'.format(
+            space_output['spaces'][space_name].keys()[0], subnet_cidr))
+    else:
+        log.info('New added space {} has been validated.'.format(space_name))
+
+
+def assert_app_status(client, charm_name, expected):
+    # Run juju status --format json
+    log.info('Checking current status of app {}.'.format(charm_name))
+    status_output = json.loads(
+        client.get_juju_output('status', '--format', 'json', include_e=False))
+    app_status = status_output['applications'][charm_name]['application-status']['current']
+
+    if app_status != expected:
+        raise JujuAssertionError(
+            'App status is incorrect. '\
+            'Found: {}\nExpected: {}\n.'.format(
+            app_status, expected))
+    else:
+        log.info('The current status of app {} is: {}; Expected: {}'.format(
+        charm_name, app_status, expected))
+
+
+def parse_args(argv):
+    """Parse all arguments."""
+    parser = argparse.ArgumentParser(
+        description="Test for lxd spaces and subnets constraints.")
+    add_basic_testing_arguments(parser)
+    return parser.parse_args(argv)
+
+
+def assess_lxd_space(client):
+    environment = os.getenv('ENV', default='parallel-aws')
```

Comment:
You mean the substrate/provider can be retrieved from client after it bootstrapped? If so then it's not what I'm checking here.
What I want here is to let the script use the right credentials and only run on AWS. I use parallel-aws to make sure it will only bootstrap on AWS, any other input will lead program to exit.